### PR TITLE
Add commonly used githooks

### PR DESCRIPTION
**Pull request by Nicoretti**
By having the the commonly used git commit hooks in a central place
they can be updated and shared among project by using https://pre-commit.com/.
This simplifies maintance und patching existing hooks.

### CODE REVIEW

File: githooks/bash/update_current_submodules_file.sh [hunk 1]
```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
```
Comment (tkilias) [inline githooks/bash/update_current_submodules_file.sh]: This is a very specific githooks which is only needed for SLC. I don't think, we need to extract it.

File: githooks/bash/create_checksums.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+CHECKSUM_DIRECTORY="$PWD/checksums"
```
Comment (tkilias) [inline githooks/bash/create_checksums.sh]: Can precommit change the working dir?

**Review by tkilias — CHANGES_REQUESTED** [review]
I guess in a second step, we would translate them to python?

File: githooks/bash/update_current_submodules_file.sh (hunk 1 above)
Comment (Nicoretti) [reply githooks/bash/update_current_submodules_file.sh]: Ok, I will remove it. I thought may be helpful if we have other git sub module based projects.

File: githooks/bash/create_checksums.sh (hunk 2 above)
Comment (Nicoretti) [reply githooks/bash/create_checksums.sh]: If you mean as configuration setting per hook, I don't think so.

**Comment by Nicoretti** [comment]
@tkilias the intention of this PR was to just add "all" commonly used commit hooks of ours and later, when used/referenced update and adjust the hooks as needed. I even would consider rewriting them in python (without dependencies).

File: githooks/bash/create_checksums.sh (hunk 2 above)
Comment (tkilias) [reply githooks/bash/create_checksums.sh]: hm, then this hook has a questionable effect as it is.

File: githooks/bash/update_current_submodules_file.sh (hunk 1 above)
Comment (tkilias) [reply githooks/bash/update_current_submodules_file.sh]: We don't want to have them. They make a lot of trouble. We are working on removing the submodule.

**Review by tkilias — APPROVED** [review]

**Pull request merged**